src/lib/storage/column_encoders/base_column_encoder.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <boost/hana/type.hpp>
+
+#include <memory>
+
+#include "storage/base_value_column.hpp"
+#include "storage/encoded_columns/column_encoding_type.hpp"
+
+#include "all_type_variant.hpp"
+#include "resolve_type.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+namespace hana = boost::hana;
+
+/**
+ * @brief Base class of all column encoders
+ *
+ * Use the column_encoder.template.hpp to add new implementations!
+ */
+class BaseColumnEncoder {
+ public:
+  /**
+   * @brief Returns true if the encoder supports the given data type.
+   */
+  virtual bool supports(DataType data_type) const = 0;
+
+  /**
+   * @brief Encodes a value column with the given data type.
+   *
+   * @return encoded column if data type is supported else nullptr
```

Comment:
Do you rely on the nullptr part? I think callers should call `supports` before and `encode` should fail if it is not supported

Reply:
Done